Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -138,6 +140,185 @@ public class PushDeploymentController : ApiController
             }
         }
 
+        [HttpPost]
+        [HttpPut]
+        public async Task<HttpResponseMessage> OnePushDeploy(
+            [FromUri] bool isAsync = false,
+            [FromUri] string author = null,
+            [FromUri] string authorEmail = null,
+            [FromUri] string deployer = OneDeploy,
+            [FromUri] string message = DefaultMessage)
+        {
+            using (_tracer.Step("OnePushDeploy"))
+            {
+                var response = Request.CreateResponse();
+
+                //Parses all query parameters into queryCollection
+                var queryCollection = Request.RequestUri.ParseQueryString();
+                var deployType = queryCollection[Constants.Type];
+                var website_framework = System.Environment.GetEnvironmentVariable(Constants.Framework).ToLower();
+                //Checks the "path" queary parameter, path root at /wwwroot TBD if need to change to /home
+                string path = queryCollection[Constants.FilePath];
+
+                //Checks the "clean" query parameter, defauts to False if no query parameter is provided
+                var cleanQuery = queryCollection[Constants.Clean];
+                bool isCleanDeployment = Convert.ToBoolean(cleanQuery);
+
+                //Checks the "async" query parameter, defaults to false if no query parameter is provided
+                var asyncQuery = queryCollection[Constants.Async];
+                isAsync = Convert.ToBoolean(asyncQuery);
```

Comment:
The function parameter isAsync already does this for you for free. Is there a reason for doing it again?